sqsconsumer_test.go

```diff
@@ -208,3 +209,102 @@ func TestSmartSQSConsumer_ReceivingMessageFailure(t *testing.T) {
 	consumer.StopConsuming(context.Background())
 
 }
+
+// TestSmartSQSConsumer_ConsumeMessageFailures tests retryable and nonretryable ConsumeMessage
+// errors.
+func TestSmartSQSConsumer_ConsumeMessageFailures(t *testing.T) {
+	// mocks
+	var ctrl = gomock.NewController(t)
+	defer ctrl.Finish()
+	mockQueue := NewMockSQSAPI(ctrl)
+	mockLogger := NewMockLogger(ctrl)
+	mockMessageConsumer := NewMockSQSMessageConsumer(ctrl)
+
+	// testBlocker is used to make this test deterministic(avoid timeouts)
+	var testBlocker sync.WaitGroup
+	var consumer = &SmartSQSConsumer{
+		Logger:          mockLogger,
+		QueueURL:        queueURL,
+		Queue:           mockQueue,
+		MessageConsumer: mockMessageConsumer,
+		NumWorkers:      10,
+		MessagePoolSize: 100,
+	}
+
+	messages := []*sqs.Message{}
+	for i := 0; i < 2; i++ {
+		messages = append(messages, defaultSQSMessage)
+	}
+
+	receiveMessageOutput := &sqs.ReceiveMessageOutput{
+		Messages: messages,
+	}
+
+	mockQueue.EXPECT().ReceiveMessage(sqsInput).Return(receiveMessageOutput, nil)
+
+	mockMessageConsumer.EXPECT().ConsumeMessage(gomock.Any(), defaultSQSMessage).Return(RetryableConsumerError{})
+	mockMessageConsumer.EXPECT().ConsumeMessage(gomock.Any(), defaultSQSMessage).Return(errors.New("a permenanet error"))
+	mockQueue.EXPECT().ChangeMessageVisibility(gomock.Any()).DoAndReturn(func(interface{}) (*sqs.DeleteMessageOutput, error) {
+		testBlocker.Done()
+		return nil, nil
+	})
+	mockQueue.EXPECT().DeleteMessage(gomock.Any()).DoAndReturn(func(interface{}) (*sqs.DeleteMessageOutput, error) {
+		testBlocker.Done()
+		return nil, nil
+	})
+	testBlocker.Add(2)
+	go consumer.StartConsuming(context.Background())
+	testBlocker.Wait()
+	consumer.StopConsuming(context.Background())
+
+}
+
+// TestSmartSQSConsumer_ConsumeMessageFailures tests retryable and nonretryable ConsumeMessage
```

Comment:
task: fix copy-pasta error on test description to something correct and meaningful.

Reply:
done